reference/entry.json

```diff
@@ -148,6 +148,246 @@
+      "patch": {
+        "operationId": "patch-subEvents",
+        "description": "Updates the given subEvents on the event with the given `eventId`.\n\nAllows partial updates, omitted properties will be ignored and remain unchanged. Omitted subEvents will also remain unchanged.\n\nOnly events with calendar type `single` and `multiple` have subEvents, so only events with those calendar types support this endpoint.",
+        "tags": [
+          "Events"
+        ],
+        "responses": {
+          "204": {
+            "description": "No Content. The subEvents have been successfully updated."
+          },
+          "400": {
+            "description": "Bad Request. Possbile error types:\n\n* https://api.publiq.be/probs/uitdatabank/calendar-type-not-supported",
+            "content": {
+              "application/problem+json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Error"
+                },
+                "examples": {
+                  "Example": {
+                    "value": {
+                      "type": "https://api.publiq.be/probs/uitdatabank/calendar-type-not-supported",
+                      "title": "Calendar type not supported",
+                      "status": 400,
+                      "detail": "Not allowed to update subEvents on calendar type: \"permanent\". Only subEvents on single and multiple calendar types can be updated."
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "$ref": "#/components/responses/Unauthorized"
+          },
+          "403": {
+            "$ref": "#/components/responses/Forbidden"
+          },
+          "404": {
+            "$ref": "#/components/responses/EventNotFound"
+          }
+        },
+        "summary": "Partially update subEvents",
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "type": "array",
+                "description": "",
+                "minItems": 1,
+                "uniqueItems": true,
+                "x-examples": {
+                  "example-1": [
+                    {
+                      "id": 0,
+                      "startDate": "2021-01-18T19:35:58+01:00",
+                      "endDate": "2021-02-05T20:30:00+00:00",
+                      "status": {
+                        "type": "Unavailable",
+                        "reason": {
+                          "nl": "Omwille van preventieve coronamaatregelen is het concert geannuleerd",
+                          "en": "The concert has been canceled due to preventive corona measures"
+                        }
+                      },
+                      "bookingAvailability": {
+                        "type": "Unavailable"
+                      }
+                    }
+                  ]
+                },
+                "items": {
+                  "type": "object",
```

Comment:
This `type` always confuses me a bit. Does this indicate that the request body is an array of sub events or it's just one sub event as an array? 😅

Reply:
The body has a type `array`, and if you use that than it can have a property `items` which describes the array values. The `items` has a type `object`. So this means that the request body is an array of objects.

Reply:
It's a bit clearer in the Stoplight UI

![Screen Shot 2021-08-06 at 11 31 56](https://user-images.githubusercontent.com/959026/128490090-2a7f50b7-8703-43cc-bc22-a9aafa98ed25.png)

Reply:
Ah yes, that is clear now. It's an array of objects inside the UI. Thanks for clarifying.